stock_picking_validate_jobcan_wf/models/stock_picking.py

```diff
@@ -56,40 +57,100 @@ def get_api_key(self, config):
             api_key = "Token " + api_key
         return api_key
 
+    def _check_jobcan_wf_status(self):
+        self.ensure_one()
+        results = []
+        if not self.jobcan_wf_number:
+            raise UserError(_("Jobcan Workflow Number missing: %s") % (self.name))
+        try:
+            params = {"id": self.jobcan_wf_number}
+            response = self.make_api_call(
+                "jobcan_wf", endpoint="v2/requests", params=params
+            )
+            response.raise_for_status()
+            results = response.json().get("results", [])
+        except Exception as e:
+            _logger.error(
+                "API call failed for picking %s with error: %s", self.id, str(e)
+            )
+            raise UserError(_("API call failed: {}".format(str(e)))) from e
+        if not results:
+            raise UserError(
+                _(
+                    "Specified Jobcan Workflow Number '%(jobcan_wf_number)s' does "
+                    "not exist: %(pick_name)s",
+                    jobcan_wf_number=self.jobcan_wf_number,
+                    pick_name=self.name,
+                )
+            )
+        if results[0].get("status") != "completed":
+            raise UserError(
+                _(
+                    "Jobcan workflow '%(jobcan_wf_number)s' is not completed: "
+                    "%(pick_name)s",
+                    jobcan_wf_number=self.jobcan_wf_number,
+                    pick_name=self.name,
+                )
+            )
+
+    @api.model
+    def _get_or_create_channel(self):
+        channel_name = f"picking_{self.id}_notifications"
+        channel = self.env["mail.channel"].search(
+            [("name", "=", channel_name)], limit=1
+        )
+        if not channel:
+            channel = self.env["mail.channel"].create(
+                {
+                    "name": channel_name,
+                    "channel_type": "chat",
+                    "channel_partner_ids": [(4, self.user_id.partner_id.id)],
+                }
+            )
+        return channel
```

Comment:
@WangTKurata This will end up cluttering the discuss screen as errors accumulate.  I suppose we don't need to create a channel per picking as long as the person in charge of the picking is notified?

We might as well do something similar to this: https://github.com/oca/ocb/blob/541b400/addons/mail_bot/models/res_users.py#L25-L38

Reply:
Yes. Notified to the person in charge is OK.

Reply:
@AungKoKoLin1997 Can you please follow up on this to adjust the cost, removing the channel creation logic?